ocean_provider/utils/datatoken.py

```diff
@@ -65,21 +65,29 @@ def verify_order_tx(
 
     # check provider fees
     datatoken_contract = get_datatoken_contract(web3, datatoken_address)
+    provider_fee_order_log = None
     provider_fee_event_logs = datatoken_contract.events.ProviderFee().processReceipt(
         tx_receipt, errors=DISCARD
     )
+    # search in all provider_fee events until we have a match. if not, we don't have a valid event
+    # also, make sure that somebody is not spoofing provider fee event from another datatoken
+    for provider_fees_logs in provider_fee_event_logs:
+        try:
+            provider_data = json.loads(provider_fees_logs.args.providerData)
+            if (
+                provider_data["dt"].lower() == datatoken_address.lower()
+                and provider_data["id"].lower() == service.id.lower()
+            ):
+                provider_fee_order_log = provider_fees_logs
+
+        except:
```

Comment:
please catch the json encoding error here separately and raise the appropriate AssertionError

Reply:
also, can you please wrap just the problematic function using try/except? The second condition should be separate

Reply:
why ?
the logic is the following:
 - iterate over existing events
 - if event has a malformed json (or none at all), or if dt key does not exists or if id key does not exists or if dt address is not the expecting one or ....  , then this is not what we are looking for.

I don't care why it fails, that why there is only one try/catch..  as long as something failed, this event is not what we are looking for